### PR TITLE
feat(elements): Initial continue/verification paths

### PR DESCRIPTION
**Pull request by tmilewski**
### Machine Example:

https://stately.ai/registry/editor/a93475c7-595e-488b-903d-7604fb11a92e?mode=design&machineId=ee4b9181-4479-4b71-9b0f-b4b0167f35ef

### New

- Initial 'continue' flows
- Initial 'verification' flows

### Changed

- Consolidate `prepare*Verification` methods to `prepareVerification`
- Consolidate `attempt*Verification` methods to `attemptVerification`
- Simplify Clerk loading flow by using callback events
- Remove `resource` from context and directly call `clerk.client.signUp` (removes the need to constantly keep `resource` in sync with `signUp`)

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/elements/src/internals/machines/sign-up/sign-up.actors.ts [hunk 1]
```diff
@@ -65,60 +72,73 @@ export const authenticateWithSignUpRedirect = fromPromise<void, AuthenticateWith
...
-      ? {
-          password: password.value as string,
-        }
-      : {}),
+function fieldsToSignUpParams<T extends SignUpCreateParams | SignUpUpdateParams>(
+  fields: FormFields,
+): Pick<T, SignUpAdditionalKeys> {
+  const params: SignUpUpdateParams = {};
+
+  // TODO: Determine what takes priority
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-up/sign-up.actors.ts:121]: Not sure I understand that comment (I know it was there before) - isn't it obsolete now?

File: packages/elements/src/internals/machines/sign-up/sign-up.actors.ts [hunk 2]
```diff
@@ -65,60 +72,73 @@ export const authenticateWithSignUpRedirect = fromPromise<void, AuthenticateWith
...
   });
+
+  return params;
+}
+
+export type CreateSignUpInput = WithClerk<{ fields: FormFields }>;
+export type UpdateSignUpInput = WithClerk<{ fields: FormFields }>;
+
+export const createSignUp = fromPromise<SignUpResource, CreateSignUpInput>(async ({ input: { clerk, fields } }) => {
+  const params = fieldsToSignUpParams(fields);
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-up/sign-up.actors.ts:136]: Like the simplification of this ✨

File: packages/elements/src/internals/machines/sign-up/sign-up.machine.ts [hunk 3]
```diff
@@ -376,113 +369,162 @@ export const SignUpMachine = setup({
...
                 id: 'prepareEmailAddressLinkVerification',
-                src: 'prepareEmailAddressVerification',
+                src: 'prepareVerification',
                 input: ({ context }) => ({
                   clerk: context.clerk,
                   params: {
                     strategy: 'email_link',
                     redirectUrl: context.clerk.buildAfterSignUpUrl(),
                   },
+                  skipIfVerified: 'emailAddress',
```
Comment (LekoArts) [inline packages/elements/src/internals/machines/sign-up/sign-up.machine.ts:522]: Nice idea to move it to the helper function 👍

**Review by LekoArts — APPROVED** [review]
LGTM, only one non-blocking comment

File: packages/elements/src/internals/machines/sign-up/sign-up.actors.ts (hunk 1 above)
Comment (tmilewski) [reply packages/elements/src/internals/machines/sign-up/sign-up.actors.ts:121]: I believe that you're right. I'll remove in my next PR so-as not to deal with any odd rebasing.

**Pull request merged**